sympy/core/tests/test_subs.py

```diff
@@ -334,7 +334,7 @@ def test_subs_noncommutative():
     assert (x**(3*someint + 3)).subs(x**someint, L) == L**3*x**3
     assert (x**(3*someint)).subs(x**(2*someint), L) == L * x**someint
     assert (x**(4*someint)).subs(x**(2*someint), L) == L**2
-    assert (x**(4*someint + 1)).subs(x**(2*someint), L) == L**2 * x
+    assert (x**(4*someint + 1)).subs(x**(2*someint), L) == L**2 * x**1.0
```

Comment:
Why 1.0 term is introduced here?

Reply:
@ylemkimon this was the case that failed the first commit.
Currently( master branch) the result is `L**2 * x`, which fails(on this branch) because x is not x**1.
So in the result `x**1` should be used rather than `x`.
But it turns out that if I use `x**1` the test fails, so I used `x**1.0`. I am not sure why this happened.

Reply:
My guess is that the `Pow._eval_subs` (or similar) is evaluating the exponent. I don't think the answer should have a `1.0` in the exponent of x.

Reply:
If `assert (x**(4*someint + 1)).subs(x**(2*someint), L) == L**2 * x` fails then that means the `x` is not really `x` but is likely Pow(x, 1, evaluate=False)`. So if this line, `new_l.append(Pow(self.base, Add(*o_al), evaluate=False))` is still present in master (as it is in live.sympy) then it should be

```
expo = Add(*o_al)
new_l.append(Pow(self.base, expo, evaluate=False) if expo != 1 else self.base)
```

Reply:
@ylemkimon , the `1.0` is a hack to get the x (which I think is actually an unevaluated Pow with exponent of 1) to compare equal to a power. The only way to write a power with an exponent of 1 without using the `evaluate` keyword is to write it (now) as `x**1.0`. But...as I say in the comment above, this should not be necessary -- the routine should simply not return an unevaluated power when the exponent is 1.